friends/friendsController.js

```diff
@@ -0,0 +1,160 @@
+const router = require('express').Router();
+const friend = require('./friendsModel');
+
+// methods
+const errorMessage = (status, message, res) => {
+  res.status(status).json({
+    errorMessage: message
+  })
+  return;
+}
+
+router
+  .route('/')
+  .get((req, res) => {
+    friend.find()
+      .then(friends => {
+        // if (!friends.friends) {
+        //   errorMessage(404, 'friends could not be found', res);
+        // }
+        res.status(200).json(friends);
+      })
+      .catch(err => res.status(500).json({
+        error: 'Error making friends'
+      }));
+  })
+  .post((req, res) => {
+    const {
+      firstName,
+      lastName,
+      age
+    } = req.body;
+    const newFriend = new Friend({
+      firstName,
+      lastName,
+      age
+    });
+    if (!firstName || !lastName || !age) {
+      res.status(400).json({
+        errorMessage: "Please provide firstName, lastName and age for the friend."
+      })
+    } else if (isNaN(age) || age < 1 || age > 120) {
+      res.status(400).json({
+        errorMessage: "Age must be a number between 1 and 120"
+      })
+    } else {
+      newFriend
+        .save()
+        .then(savedFriend => {
+          res.status(201).json(savedFriend);
+        })
+        .catch(error => {
+          res.status(422).json({
+            error: error
+          });
+        });
+    }
+  });
+
```

Comment:
I don't know if you caught this in the CS10 channel, but Dre'Sean shared a cool trick: if you use ({ error: error.message }) you get a more descriptive error message in your json response (or console.log statement).